flow/aggregates.py

```diff
@@ -303,12 +302,12 @@ class _AggregatesStore:
     """
     def __init__(self, aggregator, project):
         self._aggregator = aggregator
-
+        self._project = project
         # We need to create a _MakeAggregate instance associated with this object
         # in order to generate aggregates whenever necessary.
         self._make_aggregates = _MakeAggregates(
             aggregator._aggregator_function, aggregator._sort_by,
-            aggregator._reverse_order, aggregator._select, project)
+            aggregator._reverse_order, aggregator._select)
```

Comment:
We could use a classmethod for `_MakeAggregates` that takes the `aggregator` and handles this setting internally.